lib/exercises.rb

```diff
@@ -2,19 +2,65 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O((n * m), where n is the number of elements in strings and m is the average length of an element in strings)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  anagram_hashes = []
+  strings.each do |string|
+    anagram_match = false
+    new_hash = {}
+    string.each_char do |n|
+      if new_hash[n]
+        new_hash[n] += 1
+      else
+        new_hash[n] = 1
+      end 
+    end
+    anagram_hashes.each do |stored_hash|
+      if stored_hash["hash"] == new_hash 
+        stored_hash["words"] << string
+        anagram_match = true
+      end 
+    end 
+    if anagram_match == false
+      anagram_hashes << {
+        "words" => [string],
+        "hash" => new_hash
+      }
+    end 
```

Comment:
Why not make `anagram_hashes` a hash instead of an array and use `new_hash` as a key with the words as the value.

The way you have this `anagram_hashes` is likely to be O(n), and since it's nested in `strings.each`... this solution is O(n^2).